### ISSUE TITLE
integration::workers:workers is flaky

### ISSUE THREAD

**Issue by littledivy** [denoland/deno#11344]
Noticed this happen a few times earlier. Example CI runs -
* https://github.com/denoland/deno/runs/3021774831#step:34:2942
* https://github.com/denoland/deno/pull/11349/checks?check_run_id=3035033822#step:34:2073
* https://github.com/denoland/deno/pull/11341/checks?check_run_id=3025922808#step:34:2942
* https://github.com/denoland/deno/runs/3036683371#step:34:2946
* https://github.com/denoland/deno/pull/11413/checks?check_run_id=3078749844#step:33:3083

Error:
```shell
test Worker with top-level-await ...
test result: FAILED. 24 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (8151ms)

error: Module evaluation is still pending but there are no pending ops or dynamic imports. This situation is often caused by unresolved promise.
test integration::worker::workers ... FAILED

failures:

---- integration::worker::workers stdout ----
Error: thread 'integration::worker::workers' panicked at 'assertion failed: status.success()', cli\tests\integration\worker_tests.rs:23:3
error: test failed, to rerun pass '-p deno --test integration_tests'
stack backtrace:
   0:     0x7ff77e9ce7e3 - std::backtrace_rs::backtrace::dbghelp::trace
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
   1:     0x7ff77e9ce7e3 - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff77e9ce7e3 - std::sys_common::backtrace::_print_fmt
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys_common\backtrace.rs:67
   3:     0x7ff77e9ce7e3 - std::sys_common::backtrace::_print::{{impl}}::fmt
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys_common\backtrace.rs:46
   4:     0x7ff77e8597fc - core::fmt::write
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\core\src\fmt\mod.rs:1094
   5:     0x7ff77e9d38d8 - std::io::Write::write_fmt<alloc::vec::Vec<u8, alloc::alloc::Global>>
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\io\mod.rs:1584
   6:     0x7ff77e9cd08e - std::sys_common::backtrace::_print
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys_common\backtrace.rs:49
   7:     0x7ff77e9cd08e - std::sys_common::backtrace::print
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys_common\backtrace.rs:36
   8:     0x7ff77e9cd08e - std::panicking::default_hook::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\panicking.rs:208
   9:     0x7ff77e9cc56d - std::panicking::default_hook
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\panicking.rs:222
  10:     0x7ff77e9cc56d - std::panicking::rust_panic_with_hook
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\panicking.rs:591
  11:     0x7ff77e9d47ab - std::panicking::begin_panic_handler::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\panicking.rs:495
  12:     0x7ff77e9d472f - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys_common\backtrace.rs:141
  13:     0x7ff77e9d46cf - std::panicking::begin_panic_handler
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\panicking.rs:493
  14:     0x7ff77ec282d0 - core::panicking::panic_fmt
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\core\src\panicking.rs:92
  15:     0x7ff77ec2821c - core::panicking::panic
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\core\src\panicking.rs:50
  16:     0x7ff77e7621f8 - main
  17:     0x7ff77ea1a7e6 - core::ops::function::FnOnce::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\core\src\ops\function.rs:227
  18:     0x7ff77ea1a7e6 - test::__rust_begin_short_backtrace<fn()>
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\test\src\lib.rs:577
  19:     0x7ff77ea1aab6 - alloc::boxed::{{impl}}::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\alloc\src\boxed.rs:1546
  20:     0x7ff77ea1aab6 - std::panic::{{impl}}::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panic.rs:344
  21:     0x7ff77ea1aab6 - std::panicking::try::do_call
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panicking.rs:379
  22:     0x7ff77ea1aab6 - std::panicking::try
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panicking.rs:343
  23:     0x7ff77ea1aab6 - std::panic::catch_unwind
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panic.rs:431
  24:     0x7ff77ea1aab6 - test::run_test_in_process
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\test\src\lib.rs:600
  25:     0x7ff77ea1aab6 - test::run_test::run_test_inner::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\test\src\lib.rs:492
  26:     0x7ff77ea1caa1 - test::run_test::run_test_inner::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\test\src\lib.rs:519
  27:     0x7ff77ea1caa1 - std::sys_common::backtrace::__rust_begin_short_backtrace<closure-1,tuple<>>
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\sys_common\backtrace.rs:125
  28:     0x7ff77ea1c82a - std::thread::{{impl}}::spawn_unchecked::{{closure}}::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\thread\mod.rs:481
  29:     0x7ff77ea1c82a - std::panic::{{impl}}::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panic.rs:344
  30:     0x7ff77ea1c82a - std::panicking::try::do_call
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panicking.rs:379
  31:     0x7ff77ea1c82a - std::panicking::try
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panicking.rs:343
  32:     0x7ff77ea1c82a - std::panic::catch_unwind
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\panic.rs:431
  33:     0x7ff77ea1c82a - std::thread::{{impl}}::spawn_unchecked::{{closure}}
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\std\src\thread\mod.rs:480
  34:     0x7ff77ea1c82a - core::ops::function::FnOnce::call_once<closure-0,tuple<>>
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\core\src\ops\function.rs:227
  35:     0x7ff77e9ed18a - alloc::boxed::{{impl}}::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\alloc\src\boxed.rs:1546
  36:     0x7ff77e9ed18a - alloc::boxed::{{impl}}::call_once
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\library\alloc\src\boxed.rs:1546
  37:     0x7ff77e9ed18a - std::sys::windows::thread::{{impl}}::new::thread_start
                               at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b\/library\std\src\sys\windows\thread.rs:56
  38:     0x7ffeb1ba7974 - BaseThreadInitThunk
  39:     0x7ffeb467a2f1 - RtlUserThreadStart


failures:
    integration::worker::workers

test result: FAILED. 457 passed; 1 failed; 14 ignored; 0 measured; 0 filtered out; finished in 369.06s
```

**Comment by dsherret** [comment]
Maybe it's the same cause as what's going on in #11128?

**Comment by bartlomieju** [comment]
I believe this was fixed in #11540
